src/app/feature/shared-data/providers/firebase.ts

```diff
@@ -146,8 +146,14 @@ export class FirebaseDataProvider implements SharedDataProviderBase {
   private collectionToObservable(
     options: AddCollectionSnapshotListenerOptions
   ): Observable<ISharedDataCollection[]> {
+    // Timeout for server data to be received before falling back to cache
+    const TIMEOUT = 2000;
```

Comment:
Blocking
Having an arbitrary timeout to receive server data is a really bad pattern - large data payloads, flaky connection, slow servers, distanct server locations etc. can all play a factor into why a credible request may still take a considerable amount of time. This may limit the ability to ever receive live data from server, breaking functionality